ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -0,0 +1,228 @@
+    Pair<Long, NavigableMap<String, Long>> logSegmentValidSizeResult = getValidDataSizeByLogSegment(timeRange);
+    Long totalValidSize = 0L;
+    for (Long value : logSegmentValidSizeResult.getSecond().values()) {
+      totalValidSize += value;
+    }
+    return new Pair<>(logSegmentValidSizeResult.getFirst(), totalValidSize);
+  }
+
+  /**
+   * Gets the size of valid data at a particular point in time for all log segments. The caller specifies a reference time and
+   * acceptable resolution for the stats in the form of a {@link TimeRange}. The store will return data for a point in time within
+   * the specified range.
+   * The following data are considered as valid data for this API:
+   * 1. PUT with no expiry and no corresponding DELETE
+   * 2. PUT expiring at t_exp but t_ref < t_exp
+   * 3. PUT with DELETE at time t_delete but t_ref < t_delete
+   * 4. DELETE record
+   * @param timeRange the reference {@link TimeRange} at which the data is requested. Defines both the reference time and the acceptable resolution.
+   * @return a {@link Pair} whose first element is the time at which stats was collected (in ms) and whose second element is the valid data
+   * size for each segment in the form of a {@link NavigableMap} of segment names to valid data sizes.
+   */
+  Pair<Long, NavigableMap<String, Long>> getValidDataSizeByLogSegment(TimeRange timeRange) throws StoreException {
+    long deleteReferenceTimeInMs = timeRange.getEndTimeInMs();
+    long expirationReferenceTimeInMs = time.milliseconds();
+    NavigableMap<String, Long> validSizePerLogSegment =
+        collectValidDataSizeByLogSegment(deleteReferenceTimeInMs, expirationReferenceTimeInMs);
+    return new Pair<>(deleteReferenceTimeInMs, validSizePerLogSegment);
+  }
+
+  /**
+   * Gets the size of valid data for all serviceIds and their containerIds as of now (the time when the API is called).
+   * The following data are considered as valid data for this API:
+   * 1. PUT with no expiry and no corresponding DELETE
+   * 2. PUT expiring at t_exp but t_ref < t_exp
+   * 3. PUT with DELETE at time t_delete but t_ref < t_delete
+   * @return the valid data size for each container in the form of a nested {@link Map} of serviceIds to another map of
+   * containerIds to valid data size.
+   */
+  Map<String, Map<String, Long>> getValidDataSizeByContainer() throws StoreException {
+    long deleteAndExpirationRefTimeInMs = time.milliseconds();
+    return collectValidDataSizeByContainer(deleteAndExpirationRefTimeInMs);
+  }
+
+  /**
+   * Walk through the entire index and collect valid data size information per container (size of delete records not included).
+   * @param deleteAndExpirationRefTimeInMs the reference time in ms until which deletes and expiration are relevant
+   * @return a nested {@link Map} of serviceId to containerId to valid data size
+   */
+  private Map<String, Map<String, Long>> collectValidDataSizeByContainer(long deleteAndExpirationRefTimeInMs)
```

Comment:
Having it to be a function of its own makes it more explicit about what it's doing.